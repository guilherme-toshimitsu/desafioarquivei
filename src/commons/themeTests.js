import React from "react";
import { ThemeProvider } from "styled-components";
import { render } from "@testing-library/react";
import * as Redux from "react-redux";
import thunk from "redux-thunk";
import theme from "./theme";
import store from "@store";

const renderWithTheme = (children) => {
  return render(<ThemeProvider theme={theme}>{children}</ThemeProvider>);
};

const renderWithThemeAndRedux = (children) => {
  return render(
    <ThemeProvider theme={theme}>
      <Redux.Provider store={store}>{children}</Redux.Provider>
    </ThemeProvider>
  );
};

export { renderWithTheme, renderWithThemeAndRedux };
