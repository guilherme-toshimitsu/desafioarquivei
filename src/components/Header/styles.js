import styled from "styled-components";
import { breakpoints } from "@commons/theme";

const HeaderContainer = styled.header`
  display: flex;
  justify-content: space-between;
  top: 0;
  width: 100%;
  background-color: ${({ theme }) => theme.colors.main};
  height: 60px;
  cursor: pointer;

  ${breakpoints.sm} {
    height: 40px;
    display: flex;
    flex-direction: column-reverse;
  }
`;

const Img = styled.img`
  margin: auto 0;
  height: 50px;
  ${breakpoints.sm} {
    display: none;
  }
`;

const HeaderButton = styled.button`
  width: 100px;
  height: 60px;
  border: 0px;
  color: white;
  font-size: 24px;
  background-color: ${({ theme }) => theme.colors.main};
`;

export { HeaderContainer, Img, HeaderButton };
