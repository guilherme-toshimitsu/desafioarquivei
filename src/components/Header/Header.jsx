import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import Link from "next/link";
import Router from "next/router";
import { actions } from "@store/User";
import { HeaderContainer, Img, SubContainer } from "./styles";
import Button from "@components/Button";

const Header = () => {
  const dispatch = useDispatch();
  const { auth, isLoading, email } = useSelector((state) => state.userStore);

  const redirectToHome = () => {
    if (auth) {
      Router.push("/home");
    } else {
      Router.push("/login");
    }
  };

  return (
    <HeaderContainer>
      <Img
        src={"/assets/logo-white-small.png"}
        onClick={() => redirectToHome()}
      />
      {/* <SubContainer> */}
      {auth ? (
        <>
          {email}
          <Button onClick={() => dispatch(actions.logout())}>Logout</Button>
        </>
      ) : (
        <Button onClick={() => Router.push("/login")}>Login</Button>
      )}
      {/* </SubContainer> */}
    </HeaderContainer>
  );
};

export default Header;
