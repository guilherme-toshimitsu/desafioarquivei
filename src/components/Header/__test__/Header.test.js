import React from "react";
import Header from "..";
import { renderWithThemeAndRedux } from "@commons/themeTests";

describe("<Header />", () => {
  it("creates a Header component", () => {
    const wrapper = renderWithThemeAndRedux(<Header />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
