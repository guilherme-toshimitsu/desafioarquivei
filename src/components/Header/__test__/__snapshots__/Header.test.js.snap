// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> creates a Header component 1`] = `
<DocumentFragment>
  <header
    class="styles__HeaderContainer-xa09ss-0 eooKdB"
  >
    <img
      class="styles__Img-xa09ss-1 jcOUEY"
      src="/assets/logo-white-small.png"
    />
    <button
      class="Button__StyledButton-pzevnv-0 iLzPQp"
    >
      Login
    </button>
  </header>
</DocumentFragment>
`;
